pkg/cmd/ssh.go

```diff
@@ -138,12 +122,19 @@ func getSSHKeypair(shoot *gardencorev1alpha1.Shoot) *v1.Secret {
 }
 
 // printNodeNames print all nodes in k8s cluster
-func printNodeNames(kindIP string) {
+func printNodeNames(kindIP string, shoot *gardencorev1alpha1.Shoot) (err error) {
+	if shoot.Status.IsHibernated {
+		return fmt.Errorf("shoot %q is hibernated", shoot.Name)
+	}
+
 	machines := getMachines().Items
```

Comment:
You can pass the Shoot name to `getMachines` to prevent reading it once again from the target.